src/RepositoryApiClient.cs

```diff
@@ -73,12 +75,27 @@ public static IRepositoryApiClient CreateFromHttpRequestHandler(IHttpRequestHand
         /// </summary>
         /// <param name="servicePrincipalKey"></param>
         /// <param name="accessKey"></param>
-        /// <param name="baseUrlDebug">Optional override for the Laserfiche repository API base url.</param>
+        /// <param name="baseUrlDebug"> Optional override for the Laserfiche repository API base url.</param>
         /// <returns></returns>
         public static IRepositoryApiClient CreateFromAccessKey(string servicePrincipalKey, AccessKey accessKey, string baseUrlDebug = null)
         {
             var httpRequestHandler = new OAuthClientCredentialsHandler(servicePrincipalKey, accessKey);
             return CreateFromHttpRequestHandler(httpRequestHandler, baseUrlDebug);
+        }
+
+        /// <summary>
+        /// Create a Laserfiche repository client that will use LFDS username and password to get access tokens.
+        /// </summary>
+        /// <param name="username"></param>
+        /// <param name="password"></param>
+        /// <param name="organization"></param>
+        /// <param name="repoID"></param>
+        /// <param name="baseUrl"></param>
+        /// <returns></returns>
+        public static IRepositoryApiClient CreateFromLfdsUsernamePassword(string username, string password, string organization, string repoID, string baseUrl)
+        {
+            var httpRequestHandler = new LfdsUsernamePasswordHandler(username, password, organization, repoID, baseUrl);
+            return CreateFromHttpRequestHandler(httpRequestHandler);

```

Comment:
i think you could call `CreateFromHttpRequestHandler` with the second `baseUrlDebug` parameter as `baseUrl` and then you won't need any changes to the `RepositoryApiClientHandler`, since that will set the base url correctly

Reply:
Done. And i think i can remove the `SelfHostedBaseUrl` property from `BeforeSendResult` class.